beacon_node/lighthouse_network/src/behaviour/mod.rs

```diff
@@ -1125,6 +1159,13 @@ impl<TSpec: EthSpec> Behaviour<TSpec> {
             self.peer_manager.update_gossipsub_scores(&self.gossipsub);
         }
 
+        // poll the gossipsub cache to clear expired messages
+        while let Poll::Ready(Some(result)) = self.gossip_cache.poll_next_unpin(cx) {
```

Comment:
It might be good to log a warning or have a metric so we know how many messages we are just dropping because we didn't find peers.